pkg/executor/adapter.go

```diff
@@ -716,11 +716,20 @@ func (a *ExecStmt) handleForeignKeyCascade(ctx context.Context, fkc *FKCascadeEx
 			terror.Log(exec.Close(e))
 			return err
 		}
-		err = exec.Next(ctx, e, exec.NewFirstChunk(e))
-		if err != nil {
-			return err
+		func() { // Recover panic in exec.Next() call.
```

Comment:
Can we add recover in `func Next(ctx context.Context, e Executor, req *chunk.Chunk) error {`, and invoke `Open/Next/Close` instead of `exec.Open/Next/Close` here?